modules/targetAcquisition/taxi/Taxi.py

```diff
@@ -0,0 +1,331 @@
+                bbox = self.bbox[self.nextUncheckedID]
+                print(f"Switching to track and NOT recalibrating. bbox: {bbox}")
+            # Else update distance with recalibrated bbox
+            else:
+                print("Switching to track and recalibrating")
+                bbox, _ = self.find_overlapped_bbox()
+                print(f"New bbox after recalibration: {bbox}")
+
+            # If recalibration failed to find the new box
+            if bbox == None:
+                print("Failed to find a bbox")
+
+            # If recalibration found the new box, recalculate distance and carry on moving
+            else:
+                print("At least one bbox is found")
+                bboxReformat = (bbox[0][0], bbox[0][1],
+                                bbox[1][0] - bbox[0][0], bbox[1][1] - bbox[0][1])
+                # Note: must create the tracker here instead of __init__, else can't switch freely between track & box
+                self.tracker = cv2.TrackerKCF_create()
+                # Initialize tracker with first frame and bounding box
+                self.tracker.init(self.frame, bboxReformat)
+                print("Tracker created and initialized")
+                # Calculate distance from tracked bbox
+                self.distanceFromBox = self.calculate_distance(bbox)
+                print(f"Calculated distance: {self.distanceFromBox}")
+                self.state = "TRACK"
+                print("Set variable self.state to TRACK")
+        else:
+            print("Error: invalid state selected")
+
+    def main(self, cap):
+        """
+        Main operations: getting camera input and passing the image to appropriate methods
+
+        Parameters
+        ----------
+        cap : np.ndarray
+            the current captured rgb image from the camera
+        """
+
+        self.frame = cap
+        if self.state == "BOX":
+            self.bbox = self.yolo.detect_boxes(self.frame)
+            for (topLeft, botRight) in self.bbox:
+                cv2.rectangle(self.frame, topLeft,
+                                botRight, (0, 0, 255), 2)
+
+            # If initial run, expect 5 boxes
+            # If recalibrating, expect at least 1 box (less accurate but there's no other way)
+            if not self.recalibrate:
+                expCount = self.expectedCount
+            else:
+                print("Recalibrating in BOX mode. Expecting 1 box")
+                expCount = 1
+
+            # YOLO can't move on until all 5 boxes stay in frame consistently/continuously for more than frameCount frames
+            if len(self.bbox) >= expCount:
+                self.frameCount += 1
+            else:
+                frameCount = 0
+
+            if frameCount == self.numStableFrames:
+                print("Got the required number of stable frames")
+                self.set_state("TRACK")
+                frameCount = 0
+
+            self.totalWait += 1
+            if self.totalWait > 250:
```

Comment:
Double check if this is a good number to stick with: at 15fps, 250 frames is 17 seconds.

Reply:
10 seconds sounds like a nicer round number maybe?